### PR TITLE
LPS-198585 Refactor SingleSelect component to use Clay Picker

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPS-198585

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java [hunk 1]
```diff
@@ -37,6 +37,8 @@ public static List<Map<String, String>> getObjectFieldBusinessTypeMaps(
 					objectFieldBusinessType.getDescription(locale)
 				).put(
 					"label", objectFieldBusinessType.getLabel(locale)
+				).put(
+					"value", objectFieldBusinessType.getName()
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java:41]: @danvb112 Following the existing pattern, you should call this property "name", based on the name of the method returning it.

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsValidationsDisplayContext.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ public List<Map<String, String>> getObjectValidationRuleEngines() {
 					objectValidationRuleEngine.getLabel(
 						objectRequestHelper.getLocale())
 				).put(
-					"name", objectValidationRuleEngine.getKey()
+					"value", objectValidationRuleEngine.getKey()
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsValidationsDisplayContext.java:99]: @danvb112 Following the existing pattern, this property should be called "key".

**Review by naraandrade — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java (hunk 1 above)
Comment (danvb112) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java:41]: Hey @naraandrade, Picker requires a key called `value` in the JSON that we are sending to it, so I need this key, what I could do is name this key here as `name`, and in the frontend take this json and put the key `value` there with the same value, what do you think?

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsValidationsDisplayContext.java (hunk 2 above)
Comment (danvb112) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsValidationsDisplayContext.java:99]: @naraandrade, same comment as above

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java (hunk 1 above)
Comment (naraandrade) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java:41]: > name this key here as name, and in the frontend take this json and put the key value there with the same value
It's better to follow that approach. We should keep the backend patterns.

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java (hunk 1 above)
Comment (danvb112) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/util/ObjectFieldBusinessTypeUtil.java]: Thanks for the suggestion @naraandrade ! Done!

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:object

**Comment by Natanaelbenvindo** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:test:object

**Comment by andreefarias** [comment]
ci:stop

**Comment by andreefarias** [comment]
ci:stop

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:object

**Comment by timpak** [comment]
ci:test:object

**Comment by timpak** [comment]
@andreefarias I'm going to summarize what I shared on Slack here. The changes look good but we should rethink the paths we're using. Since we're using Clay, we should be able to use existing generic paths. We can go forward with this approach for now but we should revisit this in the future. 

I'm really glad you made separate commits with clear messages, it made reviewing the changes super easy!

**Comment by andreefarias** [comment]
I'll send a new PR, since Daniel is on PTO.

**Pull request closed without merging**